scalagen/shared/src/test/scala/org/scalameta/scalagen/TestExtensions.scala

```diff
@@ -0,0 +1,96 @@
+package org.scalameta.scalagen
+
+import scala.meta._
+import scala.meta.contrib._
+import org.scalameta.scalagen.generators._
+import org.scalatest._
+
+class TestExtensions extends FunSuite {
+
+  test("Expansion works") {
+    val clazz: Defn.Class = q"@StructuralToString case class Foo(x: Int, y: Int)"
```

Comment:
I would recommend making the tests as declarative as possible
```scala
check(
  "@StructuralToString case class Foo(x: Int, y: Int)",
  "<expanded code>"
)
```
In testkit we have a `DiffAssertions` trait that can be mixed into a FunSuiteLike to get `assertNoDiff(obtained: String, expected)`.

This makes the tests easier to read, failures easier to debug, and also is closer to the actual integration (without being an integration test) since you use the parser instead of quasiquotes and test the annotation expansion mechanism

Reply:
Sounds good. I'll take a look

Reply:
Is testkit published?